File: Array/FindAllNumbersDisappearedinanArray.scala
object Solution {
    def findDisappearedNumbers(nums: Array[Int]): List[Int] = {
        var list:List[Int] = List();
        val n = nums.length;
        for (ele <- nums) {
            nums((ele-1)%n) += n;
        }
        for (i <- 0 until n) {
            if (nums(i) <= n) {
                list = list:+(i+1);
            }
        }
        list;
    }
}

File: Array/FindAllNumbersDisappearedinanArray.java
class Solution {
    public List<Integer> findDisappearedNumbers(int[] nums) {
        List<Integer> list = new ArrayList();
        int n = nums.length;
        for (int i : nums) {
            nums[(i-1)%n] += n;
        } 
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] <= n) {
                list.add(i+1);
            }
        }
        return list;
    }
}
